### PR TITLE
Add portal_vapp= functionality to sheet_utils

### PR DESCRIPTION
**Pull request by netsettler**
This is a branch-to-branch PR that adds `portal_vapp=` functionality to the sheet_utils beta.

### CODE REVIEW

File: dcicutils/ff_utils.py [hunk 1]
```diff
@@ -942,41 +942,92 @@ def _get_es_metadata(uuids, es_client, filters, sources, chunk_size, auth):
...
+    portal_env = resolve_portal_env(ff_env=ff_env, portal_env=portal_env, portal_vapp=portal_vapp)
+    base_url = f"profiles/{to_camel_case(name)}.json"
+    add_on = 'frame=raw'
+    if portal_vapp:
+        full_url = f"{base_url}?{add_on}"
+        res = portal_vapp.get(full_url)
+        return get_response_json(res)
+    else:
+        auth = get_authentication_with_server(auth=key, ff_env=portal_env)
+        schema = get_metadata(obj_id=base_url, key=auth, add_on=add_on)
```
Comment (willronchetti) [inline dcicutils/ff_utils.py:998]: If you already have key, there is no need to call `get_authentication_with_server`?

File: dcicutils/ff_utils.py [hunk 2]
```diff
@@ -942,41 +942,92 @@ def _get_es_metadata(uuids, es_client, filters, sources, chunk_size, auth):
...
-    schemas: Dict[str, Dict] = get_metadata('profiles/', key=auth, add_on='frame=raw')
+    portal_env = resolve_portal_env(ff_env=ff_env, portal_env=portal_env, portal_vapp=portal_vapp)
+    base_url = 'profiles/'
+    add_on = 'frame=raw'
+    if portal_vapp:
+        full_url = f"{base_url}?{add_on}"
+        schemas: Dict[str, Dict] = portal_vapp.get(full_url)
+    else:
+        auth = get_authentication_with_server(auth=key, ff_env=portal_env)
+        schemas: Dict[str, Dict] = get_metadata(obj_id=base_url, key=auth, add_on=add_on)
```
Comment (willronchetti) [inline dcicutils/ff_utils.py:1030]: Same comment here - duplicate call to `get_authentication_with_server` when `key` present

**Review by willronchetti — APPROVED** [review]
Approving but should review what I think are duplicate calls for authentication

File: dcicutils/ff_utils.py (hunk 1 above)
Comment (netsettler) [reply dcicutils/ff_utils.py:998]: `get_authentication_with_server` only calls the server if you don't have the auth already. It's not a very good name. But also, this is how it worked before. I only added the other arm of this `if` statement.

Comment (netsettler) [reply dcicutils/ff_utils.py]: Fixed.

File: dcicutils/ff_utils.py (hunk 2 above)
Comment (netsettler) [reply dcicutils/ff_utils.py]: Fixed.

**Pull request merged**